src/dkg_hybriddkg.erl

```diff
@@ -530,112 +530,186 @@ verify_proofs({vss_ready_proofs, Proofs}, DKG = #dkg{n=N, t=T, f=F}) ->
-                            echo_proofs=Echo_Proofs,
-                            echo_counts=EchoCount,
-                            ready_counts=ReadyCounts,
-                            leader_changing=LeaderChanging,
-                            leader=Leader,
-                            leader_cap=LeaderCap,
+-spec serialize(dkg()) -> #{atom() => map() | binary()}.
+serialize(#dkg{id = Id,
+               n = N,
+               f = F,
+               t = T,
+               u = U,
+               u2 = U2,
+               shares_map = SharesMap,
+               shares_results = SharesResults,
+               shares_acked = SharesAcked,
+               shares_seen = SharesSeen,
+               vss_ready_proofs = ReadyProofs,
+               echo_proofs = Echo_Proofs,
+               echo_counts = EchoCount,
+               ready_counts = ReadyCounts,
+               leader_changing = LeaderChanging,
+               leader = Leader,
+               leader_cap = LeaderCap,
+               leader_vote_counts = LeaderVoteCounts,
+               await_vss = AwaitVSS}) ->
+    PreSer = #{u => erlang_pbc:element_to_binary(U),
+               u2 => erlang_pbc:element_to_binary(U2),
+               shares_map => serialize_shares_map(SharesMap),
+               shares_results => serialize_shares_results(SharesResults)},
+    M0 = #{id => Id,
+           n => N,
+           f => F,
+           t => T,
+           shares_seen => SharesSeen,
+           shares_acked => SharesAcked,
+           vss_ready_proofs => ReadyProofs,
+           echo_proofs => Echo_Proofs,
+           echo_counts => EchoCount,
+           ready_counts => ReadyCounts,
+           leader_changing => LeaderChanging,
+           leader => Leader,
+           leader_cap => LeaderCap,
+           leader_vote_counts => LeaderVoteCounts,
+           await_vss => AwaitVSS},
+    M = maps:map(fun(_K, Term) -> term_to_binary(Term) end, M0),
+    maps:merge(PreSer, M).
+
+-spec deserialize(serialized_dkg() | #{}, erlang_pbc:element(), fun(), fun()) -> dkg().
+deserialize(#serialized_dkg{id = Id,
+                            n = N,
+                            f = F,
+                            t = T,
+                            u = SerializedU,
+                            u2 = SerializedU2,
+                            shares_map = SerializedSharesMap,
+                            shares_results = SerializedSharesResults,
+                            shares_acked = SharesAcked,
+                            shares_seen = SharesSeen,
+                            vss_ready_proofs = ReadyProofs,
+                            echo_proofs = Echo_Proofs,
+                            echo_counts = EchoCount,
+                            ready_counts = ReadyCounts,
+                            leader_changing = LeaderChanging,
+                            leader = Leader,
+                            leader_cap = LeaderCap,
                             %% XXX: Only one element is enough?
```

Comment:
Yes, we can deserialize PBC elements simply with another element on the same curve...